### PR TITLE
docs: replace type `mixed` in `Cookie` class.

### PR DESCRIPTION
**Pull request by ping-yee**
**Description**
See #6310 

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: system/Cookie/Cookie.php [hunk 1]
```diff
@@ -96,7 +96,7 @@ class Cookie implements ArrayAccess, CloneableCookieInterface
      * Default attributes for a Cookie object. The keys here are the
      * lowercase attribute names. Do not camelCase!
      *
-     * @var array<string, mixed>
+     * @var array<string, bool|int|object|string>
```
Comment (kenjis) [inline system/Cookie/Cookie.php:99]: Can be `object`?

**Comment by kenjis** [comment]
```
Error: Parameter #2 $array of function implode expects array<string>, array<int, bool|int|object|string> given.
 ------ ---------------------------------------------------------------- 
  Line   system/Cookie/CookieStore.php                                   
 ------ ---------------------------------------------------------------- 
  146    Parameter #2 $array of function implode expects array<string>,  
         array<int, bool|int|object|string> given.                       
 ------ ---------------------------------------------------------------- 
```
https://github.com/codeigniter4/CodeIgniter4/actions/runs/4407171630/jobs/7720467402

File: system/Cookie/Cookie.php (hunk 1 above)
Comment (ping-yee) [reply system/Cookie/Cookie.php]: I have some misunderstanding of expires field before, I already remove it.

File: system/Cookie/Cookie.php [hunk 2]
```diff
@@ -195,9 +195,9 @@ public static function fromHeaderString(string $cookie, bool $raw = false)
     /**
      * Construct a new Cookie instance.
      *
-     * @param string               $name    The cookie's name
-     * @param string               $value   The cookie's value
-     * @param array<string, mixed> $options The cookie's options
+     * @param string                                $name    The cookie's name
+     * @param string                                $value   The cookie's value
+     * @param array<string, bool|int|object|string> $options The cookie's options
```
[the remark this answers is not in the document]
Comment (ping-yee) [reply system/Cookie/Cookie.php]: I already remove it.

**Comment by ping-yee** [comment]
@kenjis I have been fixed this problem of phpstan.

File: system/Cookie/Cookie.php [hunk 3]
```diff
@@ -595,8 +595,8 @@ public function offsetGet($offset)
     /**
      * Offset to set.
      *
-     * @param string $offset
-     * @param mixed  $value
+     * @param string                                  $offset
+     * @param array|bool|float|int|object|string|null $value
```
Comment (kenjis) [inline system/Cookie/Cookie.php:599]: `bool|int|string`?
But, after all, we cannot set anything.

File: system/Cookie/CookieInterface.php [hunk 4]
```diff
@@ -162,7 +162,7 @@ public function __toString();
     /**
      * Returns the array representation of the Cookie object.
      *
-     * @return array<string, mixed>
+     * @return array<string, bool|int|object|string>
```
Comment (kenjis) [inline system/Cookie/CookieInterface.php:165]: `bool|int|string`?

File: tests/system/Cookie/CookieTest.php [hunk 5]
```diff
@@ -168,7 +168,7 @@ public function testExpirationTime(): void
     /**
      * @dataProvider invalidExpiresProvider
      *
-     * @param mixed $expires
+     * @param int|object $expires
```
[the remark this answers is not in the document]
Comment (ping-yee) [reply tests/system/Cookie/CookieTest.php:171]: Does using `DateTimeInterface|int|string` is a good idea for this?
Like:
https://github.com/codeigniter4/CodeIgniter4/blob/d26b65ef6f2cca53b74be4843592f7af126addf9/system/Cookie/Cookie.php#L699

File: system/Cookie/Cookie.php (hunk 3 above)
Comment (ping-yee) [reply system/Cookie/Cookie.php]: Modified.

File: system/Cookie/CookieInterface.php (hunk 4 above)
Comment (ping-yee) [reply system/Cookie/CookieInterface.php]: Modified.

File: tests/system/Cookie/CookieTest.php (hunk 5 above)
[the remark this answers is not in the document]
Comment (kenjis) [reply tests/system/Cookie/CookieTest.php]: No. The invalidExpiresProvider provides invalid values for expires.

File: tests/system/Cookie/CookieTest.php [hunk 6]
```diff
@@ -168,7 +169,7 @@ public function testExpirationTime(): void
     /**
      * @dataProvider invalidExpiresProvider
      *
-     * @param mixed $expires
+     * @param DateTimeInterface|int|string $expires
```
Comment (kenjis) [inline tests/system/Cookie/CookieTest.php:172]:
```suggestion
     * @param bool|float|string $expires
```

Comment (ping-yee) [reply tests/system/Cookie/CookieTest.php]: Done.

**Review by kenjis — APPROVED** [review]
Thank you.

**Pull request merged**